### DISCUSSION TITLE
Laravel 9 File Upload issue.

### DISCUSSION THREAD

**Discussion by shaneburrell** [livewire/livewire#4631] [7. Bugs (archived)] [answered]
After upgrading to Laravel 9 with new flysystem I get the following error on all file uploads.

Call to undefined method League\Flysystem\Filesystem::getAdapter()
/Users/williamburrell/yoda/yodalaravel/vendor/livewire/livewire/src/GenerateSignedUploadUrl.php#19
Error

    public function forS3($file, $visibility = 'private')
    {
        $adapter = FileUploadConfiguration::storage()->getDriver()->getAdapter();

        if ($adapter instanceof CachedAdapter) {
            $adapter = $adapter->getAdapter();

**Comment by kylemilloy** [comment]
This is caused because L9 now uses `"league/flysystem-aws-s3-v3": "^3.0"` instead of `1.0` which doesn't allow access to the adapter and instead looks to require commands are sent through the driver directly.

**Reply by kylemilloy** [reply]
Also to note, in addition to using `getAdapter()` [here](https://github.com/livewire/livewire/blob/d65884564176ac8b9a6097d16758a07fbae33a5d/src/GenerateSignedUploadUrl.php#L19)

It's also used in two other places [here](https://github.com/livewire/livewire/blob/d65884564176ac8b9a6097d16758a07fbae33a5d/src/TemporaryUploadedFile.php#L86) and [here](https://github.com/livewire/livewire/blob/2f06b28be0aa0d5bf0a4598b4b571d7d1a24e5c1/src/Commands/S3CleanupCommand.php#L21). Haven't tested these yet to see their impact.

**Reply by shaneburrell** [reply]
Looks like some changes were made for L9 at least in one where it checks for method_exsists but not complete support for L9 and flysystem v2+ .

**Comment by joshhanley** [comment]
@shaneburrell and @kylemilloy thanks for reporting! Looks like I missed a few instances when making the flysystem V2+ changes.

We'll get that fixed up.

**Comment by joshhanley** [comment]
I've submitted PR #4650 with a fix for this 🙂

**Comment by joshhanley** [comment, marked as answer]
The fix has been merged and tagged in v2.10.2!

**Comment by deleugpn** [comment]
I noticed Laravel added a getClient to it's own adapter (https://github.com/laravel/framework/pull/41079). Would this allow Livewire to simplify some code?

**Comment by TiagoSilvaPereira** [comment]
Just to confirm, updating to the latest Livewire version solved this problem. Thanks!
